src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INCREASE_VALUE = 2;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this.buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= buckets.length * LOAD_FACTOR) {
+            resize();
+        }
+        if (doPut(key, value)) {
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = buckets[getIndex(key)];
+        if (currentNode != null) {
+            return getNode(getIndex(key), key).value;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldBuckets = buckets;
+        buckets = new Node[buckets.length * INCREASE_VALUE];
+
+        for (int i = 0; i < oldBuckets.length; i++) {
+            if (oldBuckets[i] != null) {
+                Node<K, V> node = oldBuckets[i];
```

Comment:
'for' loop can be replaced with 'foreach'
```suggestion
        for (Node<K, V> oldBucket : oldBuckets) {
            if (oldBucket != null) {
                Node<K, V> node = oldBucket;
```